testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/SparkTestUtils.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.utils;
+
+import com.google.common.base.Throwables;
+import io.airlift.log.Logger;
+import io.trino.tempto.query.QueryResult;
+import net.jodah.failsafe.RetryPolicy;
+
+import java.time.temporal.ChronoUnit;
+
+public final class SparkTestUtils
+{
+    private static final Logger log = Logger.get(SparkTestUtils.class);
+
+    public static final RetryPolicy<QueryResult> MIN_REPLICATION_ERROR_RETRY_POLICY = new RetryPolicy<QueryResult>()
+            .handleIf(SparkTestUtils::isMinReplicationError)
+            .withBackoff(1, 10, ChronoUnit.SECONDS)
+            .withMaxRetries(30)
+            .onRetry(event -> log.warn(event.getLastFailure(), "Query failed on attempt %d, will retry.", event.getAttemptCount()));
+
+    private static boolean isMinReplicationError(Throwable throwable)
+    {
+        return Throwables.getStackTraceAsString(throwable).contains("could only be written to 0 of the 1 minReplication");
```

Comment:
That may depend on Hadoop client version bundled.
That's probably whyb io.trino.tests.product.hive.HiveProductTest#ERROR_COMMITTING_WRITE_TO_HIVE_MATCH is an OR.

What about reusing that pattern, perhaps moving it out of the Hive-specific class (into some Hadoop-related class)?